pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -386,13 +380,17 @@ func TestDeleteNode(t *testing.T) {
 	}
 	go controller.Run(ctx)
 	controller.NodeUpdated(testutil.NewNode("node1"), nil)
-	// wait a bit to see if nothing will panic
-	time.Sleep(timeForControllerToProgress)
-	controller.taintedNodesLock.Lock()
-	if _, ok := controller.taintedNodes["node1"]; ok {
-		t.Error("Node should have been deleted from taintedNodes list")
+
+	// await until controller.taintedNodes is empty
+	err := wait.PollImmediate(time.Millisecond, time.Second, func() (bool, error) {
```

Comment:
1ms sounds like too fast. Let's leave it at 10.

Reply:
Updated